### PR TITLE
Build test tree with partial trimming

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
Some tests are not compatible with trimming. There are some that fail if they can't find things they need. There are also some that reflection-interate over test methods and will happily succeed if they don't find any test methods.

Cc @dotnet/ilc-contrib

### CODE REVIEW

**Review by sbomer — APPROVED** [review]

File: src/tests/nativeaot/Directory.Build.props [hunk 1]
```diff
@@ -6,5 +6,8 @@
     <!-- NativeAOT tests don't need this xunit Assert compat quirk.
          The tests explicitly test customer scenarios with eager type construction checks present. -->
     <EagerlyValidateTypeConstruction>false</EagerlyValidateTypeConstruction>
+
+    <!-- We expect trimming to be fully enabled in these tests -->
+    <AggressiveTrimming>true</AggressiveTrimming>
```
[the remark this answers is not in the document]
Comment (MichalStrehovsky) [reply src/tests/nativeaot/Directory.Build.props:11]:
```suggestion
    <EnableAggressiveTrimming>true</EnableAggressiveTrimming>
```

File: src/tests/Directory.Build.targets [hunk 2]
```diff
@@ -648,6 +648,10 @@
       Microsoft.NETCore.Native.targets expect IlcHostArch to be set but it doesn't have to -->
     <DisableUnsupportedError>true</DisableUnsupportedError>
 
+    <!-- The default TrimMode in the SDK is full. But trimming most of the tests would just make them fail,
+         so the default for the test tree is partial. -->
+    <TrimMode Condition="'$(AggressiveTrimming)' != 'true'">partial</TrimMode>
```
Comment (MichalStrehovsky) [inline src/tests/Directory.Build.targets:653]:
```suggestion
    <TrimMode Condition="'$(EnableAggressiveTrimming)' != 'true'">partial</TrimMode>
```

**Comment by vitek-karas** [comment]
Does this affect the explicit trimming tests some libraries added? We would not want that.

**Comment by MichalStrehovsky** [comment]
> Does this affect the explicit trimming tests some libraries added? We would not want that.

Those are under src/libraries. This is for src/tests.

**Pull request merged**